### ISSUE TITLE
OpenBSD minion returns incorrect "virtual" grain on KVM

### ISSUE THREAD

**Issue by sylgeist** [saltstack/salt#40498]
### Setup
Salt minion running on OpenBSD 6.0 hosted on a KVM hypervisor returns incorrect grains data:

```
    virtual:
        physical
```

Other OS's on the same KVM hypervisor return the correct virtual grain:

```
    virtual:
        kvm
```

### Steps to Reproduce Issue
Let me know what to pull!

### Versions Report
Salt Version:
           Salt: 2016.3.1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.24.0
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Jul 25 2016, 16:14:54)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.3
            ZMQ: 4.1.4
 
System Versions:
           dist:   
        machine: amd64
        release: 6.0
         system: OpenBSD
        version: Not Installed

**Comment by Ch3LL** [comment]
On the openbsd VM is dmidecode, lspci,  or prtdiag installed?

Also can you add the following logs and paste sanitized version of output here:

```
diff --git a/salt/grains/core.py b/salt/grains/core.py
index 867924a..8d2b1b4 100644
--- a/salt/grains/core.py
+++ b/salt/grains/core.py
@@ -550,6 +550,7 @@ def _virtual(osdata):
             continue
 
         output = ret['stdout']
+        log.info('output equals: {0}'.format(output))
         if command == "system_profiler":
             macoutput = output.lower()
             if '0x1ab8' in macoutput:
@@ -677,6 +678,7 @@ def _virtual(osdata):
     choices = ('Linux', 'OpenBSD', 'HP-UX')
     isdir = os.path.isdir
     sysctl = salt.utils.which('sysctl')
+    log.info('osdata kernel: {0}'.format(osdata['kernel']))
     if osdata['kernel'] in choices:
         if os.path.isdir('/proc'):
             try:
(END)
```

**Comment by eradman** [comment]
Okay to close? We have several changes merged that should have resolved this: #40836 and #42256

**Comment by sylgeist** [comment]
👍 thank you!
